### PR TITLE
Map Improvments

### PR DESCRIPTION
**Pull request by Malthusis**
This PR does several things:

- Add simple animated arrow sprites to show which directions you can go from a node. These are appropriately disabled if you cannot move for some reason, like being in combat.
- Add a "fog of war" type layer in front of the map. Nodes that have been fully scouted are "revealed" and have a lit zone around them.
- Clean up functions for moving around the map, generalizing ReturnHome to prep for...
- Add keyboard moving functionality. Default keys are Arrow Keys and WASD. Made a big handles `computed` in order to figure out which ways you can move and where to.

Before:
image

After:
image

### CODE REVIEW

File: src/stores/mapStore.ts [hunk 1]
```diff
@@ -161,12 +161,54 @@ export const useMapStore = defineStore('mapStuff', () => {
...
+            }
+        });
+        return handles
+    })
 
-    //TODO: actual scouted prop now so we can just check that instead
-    const isScouted = computed(() => {
-        const x = !!selectedNode.value.data ? selectedNode.value.data.killCount : "";
-        return x.gte(selectedNode.value.data.scoutThreshold);
-    });
```
Comment (Malthusis) [inline src/stores/mapStore.ts:169]: Unused, so I removed it.

File: src/components/CustomNode.vue [hunk 2]
```diff
@@ -1,21 +1,31 @@
...
-    class="node-boundary" 
-    :class="[{ 'selected-node': mapStore.selectedNode.id === props.id}, zoneClass, {'special': specialZone}]">
-        {{ data.areaName }}
+    <div :class="{'explored-area': props.data.killCount >= props.data.scoutThreshold}">
+        <div @mouseenter="nodeMouseover()" 
+            @mouseleave="nodeMouseover(true)" 
+            class="node-boundary" 
+            :class="[{ 'selected-node': isSelected}, zoneClass, {'special': specialZone}]">
+                {{ data.areaName }}
+        </div>
```
Comment (cm97878) [inline src/components/CustomNode.vue:8]: This'll need a full rework once we get an actual map background... if we want to have a fog of war, I think we're unfortunately going to need to be using a canvas and bezier curves between nodes to just remove an overlay.

File: src/stores/mapStore.ts [hunk 3]
```diff
@@ -161,12 +161,54 @@ export const useMapStore = defineStore('mapStuff', () => {
...
+        })
+        selectedNode.value.data.handles?.forEach((element: string) => {
+            let direction = element.split(",")[1]
+            switch(direction) {
+                case "1": handles.top = true;
+                case "2": handles.bottom = true;
+                case "3": handles.left = true;
+                case "4": handles.right = true;
+            }
+        });
```
Comment (cm97878) [inline src/stores/mapStore.ts:208]: Is this part used for anything? I couldn't find anything.

**Comment by cm97878** [comment]
We may wanna add a fadein/out transition on the arrows, or have them turn grey or turn to X's or something, just for better clarity

**Comment by cm97878** [comment]
All looks good aside from the part that may be redundant and possibly the suggestions. Merge away

File: src/stores/mapStore.ts (hunk 3 above)
Comment (Malthusis) [reply src/stores/mapStore.ts:208]: This is used for checking if the handles actually exist. It's used specifically in the input checker in `OvermapPanel`, so you can't move a direction if that direction is not available.

**Pull request merged**